modules/core/04-channel/keeper/upgrade.go

```diff
@@ -199,6 +199,7 @@ func (k Keeper) WriteUpgradeTryChannel(ctx sdk.Context, portID, channelID string
 
 	upgrade.Fields.Version = upgradeVersion
 
+	k.SetCounterpartyLastPacketSequence(ctx, portID, channelID, upgrade.LatestSequenceSend)
```

Comment:
this is our `Upgrade`'s last sequence send, right? I think we'll need to accept an additional arg and pass `msg.CounterpartyProposedUpgrade.LastSequenceSend` from msg server

Reply:
yes! Great catch, we'll need to add that as an arg.